test/test_optim.py

```diff
@@ -1332,6 +1349,15 @@ def test_exp_lr(self):
         scheduler = ExponentialLR(self.opt, gamma=0.9)
         self._test(scheduler, targets, epochs)
 
+    def test_poly_lr(self):
+        epochs = 10
+        power = 0.9
+        total_iters = 5
+        single_targets = [(1.0 - x / total_iters) ** power * 0.05 for x in range(total_iters)] + [0.0] * (epochs - total_iters)
+        targets = [single_targets, [x * epochs for x in single_targets]]
+        scheduler = PolynomialLR(self.opt, power=power, total_iters=total_iters)
+        self._test(scheduler, targets, epochs)
+
```

Comment:
Could this scheduler and especially its `min_lr` feature have an interaction with some of the existing schedulers? In particular I'm thinking what happens if it's combined with another scheduler using `ChainedScheduler`. It would be good to add a test that confirms the scheduler produces the expected result in such a scenario.

BTW a combo with `StepLR` is not needed here. The reason why the example PR that I sent you was doing it was because it was introducing a Warmup scheduler that could have interactions with it.

Reply:
For some reasons, `PolynomialLR` combined with another scheduler (e.g. `ConstantLR`) using `ChainedScheduler` + `min_lr` does not produce the expected values. I'll investigate more.

In the meantime, I added a test without `min_lr`.

Reply:
Sounds good. Let me know when you have more info.

BTW I would be fine to drop the `min_lr` feature here. I maintained it on my draft because it was available on one of the references but overall not many schedulers implement it. Also I believe that the functionality of `min_lr` can be implemented more effectively by using the `ChainedScheduler` to effectively set it to a minimum fixed value.

Reply:
Seems pretty reasonable. I looked at some other similar schedulers like `ExponentialLR` or `ConstantLR` and they don't have that feature.

However, the `total_steps` parameter is now more like a "Number of steps needed to reach lr = 0". That's because the polynomial decay function decreases the initial learning rate to 0 in `total_steps`.

What do you think?

Reply:
Your remark about the new interpretation of `total_steps` if you remove `min_lr` is accurate. I would add that the current description on doc would be more accurate in this case.

But please note that you shouldn't let my original draft implementation bias the way you think the scheduler should be structured. The reason I provided it was to highlight specific parts of the API of the `_LRScheduler` class and in particular how to write `get_lr()` and `_get_closed_form_lr()`. If you think there is a better and more natural way to structure it, I would love to see it.

Reply:
I agree with your implementation, that's why I didn't change it. The only doubt was about how to write the `get_lr` method, in particular how to compute the next decayed LR starting from the previous one, i.e. the current values in the optimizer (and not from the initial one), but you replied to me.

The `_get_closed_form_lr` implementation is pretty standard, I double-checked with another [source](https://www.tensorflow.org/api_docs/python/tf/compat/v1/train/polynomial_decay) and both use the same formula.

What is the reason behind two different implementations? Shouldn't the `get_lr` method compute the decayed learning rate always from the initial optimizer learning rate as in the `get_closed_form_lr`? It would make the scheduler simpler to understand and less prone to error (the polynomial function would be only one instead of two different versions like now).

Reply:
> IMO, it's not clear that the end learning rate is 0.

We can review the doc descriptions before you are done to ensure this is clear.

> What is the reason behind two different implementations?

Unfortunately I don't have the historical context on why this API is implemented this way. I don't regularly work on this code-base so there are lots of things I'm not familiar with. Perhaps one of the PyTorch Core devs can provide more info here.

Reply:
Ok. 

I will remove the `min_lr` parameter and fix the latest comments by tomorrow. Then, I think it's ready for one more round of reviews.

Reply:
> Ok.
> 
> I will remove the `min_lr` parameter and fix the latest comments by tomorrow. Then, I think it's ready for one more round of reviews.

Done.